### PR TITLE
Create `LightCurve` base class for analysis functions

### PR DESCRIPTION
**Pull request by drewoldag**
Created `LightCurve` base class that `StructureFunctionLightCurve` inherits from.
Updated references and tests.

### CODE REVIEW

File: src/lsstseries/analysis/light_curve.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        self._times = self._times[~nan_mask]
+        self._fluxes = self._fluxes[~nan_mask]
+        self._errors = self._errors[~nan_mask]
+
+    def _check_input_data_size_is_equal(self):
+        """Make sure that the three input np.arrays have the same size"""
+        if self._times.size != self._fluxes.size or self._times.size != self._errors.size:
+            raise ValueError("Input np.arrays are expected to have the same size.")
+
+    def _check_input_data_length_is_sufficient(self):
```
Comment (jeremykubica) [inline src/lsstseries/analysis/light_curve.py:44]: If the goal is to have this support more than just the structure function, then I would suggest making the minimum number of points an optional parameter. You could set the default to `MIN_OBSERVATIONS_REQUIRED_FOR_SF`.

File: src/lsstseries/analysis/light_curve.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
+from typing import Optional
+import numpy as np
+
+MIN_OBSERVATIONS_REQUIRED_FOR_SF = 3
```
Comment (jeremykubica) [inline src/lsstseries/analysis/light_curve.py:4]: Can this be moved into the subclass and passed into `LightCurve`?

File: src/lsstseries/analysis/light_curve.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
+from typing import Optional
```
Comment (jeremykubica) [inline src/lsstseries/analysis/light_curve.py:1]: Comment on the high level placement of this file (not this specific line): Would this make sense to go in top-level src directory as one of the fundamental data types?

File: src/lsstseries/analysis/light_curve.py (hunk 2 above)
Comment (drewoldag) [reply src/lsstseries/analysis/light_curve.py:4]: Yep, good call. I'll move this out of the base class and into the `StructureFunctionLightCurve` class.

File: src/lsstseries/analysis/light_curve.py (hunk 3 above)
Comment (drewoldag) [reply src/lsstseries/analysis/light_curve.py:1]: I can add a docstring for this class to explain the motivation for putting it here. The goal is for this to be a container class for use in analysis functions. So I _think_ that keeping it here makes sense. But if there are strong opinions, I'm happy to adjust.

Comment (jeremykubica) [reply src/lsstseries/analysis/light_curve.py:1]: I don't feel strongly about this. So feel free to keep it where it is

File: src/lsstseries/analysis/light_curve.py (hunk 1 above)
Comment (drewoldag) [reply src/lsstseries/analysis/light_curve.py:44]: This makes sense to me. I've added an input argument for the base class with a default of 0. The subclass will provide `MIN_OBSERVATIONS_REQUIRED_FOR_SF` (equal to 3) when calling `super().__init__(...)`. 

Thanks for the suggestion!

File: src/lsstseries/analysis/light_curve.py (hunk 3 above)
Comment (drewoldag) [reply src/lsstseries/analysis/light_curve.py]: Gotcha, for now, kept where it is.

File: src/lsstseries/analysis/light_curve.py (hunk 2 above)
Comment (drewoldag) [reply src/lsstseries/analysis/light_curve.py]: I've moved this constant to the `StructureFunctionLightCurve` class, and pass it as a parameter back to the base class which will do the array length check. Thanks for the suggestion here.

**Pull request merged**